Build failure: pkgsLLVM.libunwind

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `pkgsLLVM.libunwind`

### Build log

```
make[2]: Entering directory '/build/source/src'
  CC       os-linux.lo
  CC       dl-iterate-phdr.lo
  CC       mi/init.lo
  CC       mi/flush_cache.lo
  CC       mi/mempool.lo
  CC       mi/strerror.lo
  CC       aarch64/is_fpreg.lo
  CC       aarch64/regname.lo
  CC       mi/backtrace.lo
  CC       mi/dyn-cancel.lo
  CC       mi/dyn-info-list.lo
  CC       mi/dyn-register.lo
  CC       mi/Laddress_validator.lo
  CC       mi/Ldestroy_addr_space.lo
  CC       mi/Ldyn-extract.lo
  CC       mi/Lfind_dynamic_proc_info.lo
  CC       mi/Lget_accessors.lo
  CC       mi/Lget_fpreg.lo
  CC       mi/Lset_fpreg.lo
  CC       mi/Lget_proc_info_by_ip.lo
  CC       mi/Lget_proc_name.lo
  CC       mi/Lget_reg.lo
  CC       mi/Lput_dynamic_unwind_info.lo
  CC       mi/Lset_cache_size.lo
  CC       mi/Lset_caching_policy.lo
  CC       mi/Lset_iterate_phdr_function.lo
  CC       mi/Lset_reg.lo
  CC       mi/Lget_elf_filename.lo
  CC       aarch64/Los-linux.lo
In file included from aarch64/Los-linux.c:4:
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
   42 |         "ldr x0,  %[x0]\n\t"
      |         ^
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
aarch64/Gos-linux.c:42:9: error: inline assembly requires more registers than available
10 errors generated.
make[2]: *** [Makefile:4801: aarch64/Los-linux.lo] Error 1
make[2]: Leaving directory '/build/source/src'
make[1]: *** [Makefile:2914: all] Error 2
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:619: all-recursive] Error 1
```

### Additional context

On `aarch64-linux`, nixpkgs master cherry-picked 6234b3e818ccbcf22fafeef17584e88f8298c4e3

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@orivej

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"aarch64-linux"`
 - host os: `Linux 5.15.156, NixOS, 24.05 (Uakari), 24.05.20240418.bc279bb`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/run/current-system/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/libraries/libunwind/default.nix
+++ b/pkgs/development/libraries/libunwind/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, lib, fetchFromGitHub, autoreconfHook, xz, buildPackages }:
+{ stdenv, lib, fetchpatch, fetchFromGitHub, autoreconfHook, xz, buildPackages }:
 
 stdenv.mkDerivation rec {
   pname = "libunwind";
@@ -10,6 +10,11 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     hash = "sha256-rCFBHs6rCSnp5FEwbUR5veNNTqSQpFblAv8ebSPX0qE=";
   };
+
+  patches = lib.optional (stdenv.targetPlatform.useLLVM or false) (fetchpatch {
+    url = "https://github.com/libunwind/libunwind/pull/770/commits/a69d0f14c9e6c46e82ba6e02fcdedb2eb63b7f7f.patch";
+    hash = "sha256-9oBZimCXonNN++jJs3emp9w+q1aj3eNzvSKPgh92itA=";
+  });
 
   postPatch = if (stdenv.cc.isClang || stdenv.hostPlatform.isStatic) then ''
     substituteInPlace configure.ac --replace "-lgcc_s" ""

